### PR TITLE
[Improvement][Tools] Remove Mapper usage in upgrade tools

### LINKED ISSUE #15069: [Improvement][Tools] Remove Mapper usage in upgrade tools
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Due to the database schema might changed so we cannot use mapper in upgrade tools.
For example, if we add a new column columnA in version,3.2.0, and we want to upgrade from 2.x to 3.2.0, the upgrade tools might failed if we use mapper in upgrade code, since we cannot find the columnA in schema 2.x, and in our mapper there might exist sql like `select columnA from xx`

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request

close #15069

## Brief change log

- Remove mapper usage in upgrade tools
- Move database dialect into dao plugin module
- Move mybatis configuration in application.yaml

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Comment by ruanwenjun** [comment]
image
The one bug is a regex, this is old code, I don't want to change in this pr.

File: dolphinscheduler-tools/pom.xml [hunk 1]
```diff
@@ -130,19 +130,16 @@
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>mysql</artifactId>
-            <scope>test</scope>
```
Comment (caishunfeng) [inline dolphinscheduler-tools/pom.xml:133]: why remove this scope?

**Comment by caishunfeng** [comment]
LGTM, can you add some docs for guiding on how to write an upgrade program and why we should remove mapper?

**Review by caishunfeng — APPROVED** [review]

File: dolphinscheduler-tools/pom.xml (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-tools/pom.xml:133]: Since the scope is managed at dolphinscheduler-bom

**Pull request merged**